tests/Unit/PackagePrivate/DatabaseTermIdsAcquirerTest.php

```diff
@@ -0,0 +1,292 @@
+<?php
+
+namespace Wikibase\TermStore\MediaWiki\Tests\Unit\PackagePrivate;
+
+use PHPUnit\Framework\TestCase;
+use Wikibase\TermStore\MediaWiki\TermStoreSchemaUpdater;
+use Wikibase\TermStore\MediaWiki\PackagePrivate\DatabaseTermIdsAcquirer;
+use Wikibase\TermStore\MediaWiki\PackagePrivate\InMemoryTypeIdsAcquirer;
+use Wikimedia\Rdbms\IDatabase;
+use Wikimedia\Rdbms\DatabaseSqlite;
+
+/**
+ * @group TermStoreWithMediaWikiCore
+ */
+class DatabaseTermIdsAcquirerTest extends TestCase {
+
+	/**
+	 * @var IDatabase $db
+	 */
+	private $db;
+
+	public function setUp() {
+		$this->db = DatabaseSqlite::newStandaloneInstance( ':memory:' );
+		$this->db->sourceFile( TermStoreSchemaUpdater::getSqlFileAbsolutePath() );
+	}
+
+	public function testAcquireTermIdsReturnsArrayOfIdsForAllTerms() {
+		$typeIdsAcquirer = new InMemoryTypeIdsAcquirer();
+		$alreadyAcquiredTypeIds = $typeIdsAcquirer->acquireTypeIds(
+			[ 'label', 'description', 'alias' ]
+		);
+
+		$dbTermIdsAcquirer = new DatabaseTermIdsAcquirer(
+			$this->db,
+			$this->db,
+			$typeIdsAcquirer
+		);
+
+		$termsArray = [
+			'label' => [
+				'en' => 'same',
+				'de' => 'same',
+			],
+			'description' => [ 'en' => 'same' ],
+			'alias' => [
+				'en' => [ 'same', 'same', 'another', 'yet another' ]
+			]
+		];
+
+		$acquiredTermIds = $dbTermIdsAcquirer->acquireTermIds( $termsArray );
+
+		$this->assertInternalType( 'array', $acquiredTermIds );
+		$this->assertCount( 7, $acquiredTermIds );
+	}
+
+	public function testAcquireTermIdsStoresTermsInDatabase() {
+		$typeIdsAcquirer = new InMemoryTypeIdsAcquirer();
+		$alreadyAcquiredTypeIds = $typeIdsAcquirer->acquireTypeIds(
+			[ 'label', 'description', 'alias' ]
+		);
+
+		$dbTermIdsAcquirer = new DatabaseTermIdsAcquirer(
+			$this->db,
+			$this->db,
+			$typeIdsAcquirer
+		);
+
+		$termsArray = [
+			'label' => [
+				'en' => 'same',
+				'de' => 'same',
+			],
+			'description' => [ 'en' => 'same' ],
+			'alias' => [
+				'en' => [ 'same', 'same', 'another', 'yet another' ]
+			]
+		];
+
+		$acquiredTermIds = $dbTermIdsAcquirer->acquireTermIds( $termsArray );
+
+		$this->assertTermSArrayExistInDb( $termsArray, $alreadyAcquiredTypeIds );
+	}
+
+	public function testAcquireTermIdsStoresOnlyUniqueTexts() {
+		$typeIdsAcquirer = new InMemoryTypeIdsAcquirer();
+
+		$dbTermIdsAcquirer = new DatabaseTermIdsAcquirer(
+			$this->db,
+			$this->db,
+			$typeIdsAcquirer
+		);
+
+		$termsArray = [
+			'label' => [
+				'en' => 'same',
+				'de' => 'same',
+			],
+			'description' => [ 'en' => 'same' ],
+			'alias' => [
+				'en' => [ 'same', 'same', 'another', 'yet another' ]
+			]
+		];
+
+		$acquiredTermIds = $dbTermIdsAcquirer->acquireTermIds( $termsArray );
+
+		$this->assertEquals(
```

Comment:
Would be nicer to use `assertSame()` instead of `assertEquals()` (this also applies to two more places below).

It might also be worth it extracting a separate function, à la `$this->assertRowCount( 'wbt_text', '*' )`.